docs/pages/numberfield.js

```diff
@@ -0,0 +1,246 @@
+
+      <Example
+        id="helperText"
+        name="Example: Helper Text"
+        description={`Whenever you want to provide more information about a form field, you should use \`helperText\`.`}
+        defaultCode={`
+function Example(props) {
+  const [value, setValue] = React.useState('')
+
+  return (
+    <Box padding={2} color="white">
+      <NumberField
+        id="helperTextNumberField"
+        helperText="Digits only please, e.g. 8675309"
+        label="Phone number"
+        onChange={({ value }) => setValue(value)}
+        value={value}
+      />
+    </Box>
+  );
+}
+`}
+      />
+
+      <Example
+        id="errorMessageExample"
+        name="Example: Error message"
+        description={`
+    NumberField can display an error message.
+    Simply pass in an \`errorMessage\` when there is an error present and we will handle the rest.`}
+        defaultCode={`
+function Example(props) {
+  const [value, setValue] = React.useState('')
+
+  return (
+    <NumberField
+      id="errorMessageNumberField"
+      errorMessage={!value ? "This field can't be blank!" : null}
+      onChange={({ value }) => setValue(value)}
+      label="With an error message"
+      value={value}
+    />
+  );
+}
+`}
+      />
+
+      <Example
+        id="min-max-step example"
+        name="Example: min/max/step"
+        description={`
+    NumberField provides additional props specific to numerical input.
+
+    \`min\` and \`max\` can be used to define the acceptable bounds of the input (see the [ref example](#ref%20example) for more about using these for validation).
+
+    \`step\` determines the amount incremented or decremented when using the input's arrow buttons. Set this as a float to allow decimal input.
+  `}
+        defaultCode={`
+function Example(props) {
+  const [value1, setValue1] = React.useState('');
+  const [value2, setValue2] = React.useState('');
+
+  return (
+    <Flex direction="column" gap={2}>
+      <NumberField
+        id="minMaxStepExampleNumberField1"
+        label="Stepping in intervals of 5"
+        max={25}
+        min={5}
+        onChange={({ value }) => {
+          setValue1(value);
+        }}
+        placeholder="Use the arrow buttons to increase/decrease the input value"
+        step={5}
+        value={value1}
+      />
+      <NumberField
+        id="minMaxStepExampleNumberField2"
+        label="Stepping in intervals of 0.1"
+        max={2}
+        min={-2}
+        onChange={({ value }) => {
+          setValue2(value);
+        }}
+        placeholder="Use the arrow buttons to increase/decrease the input value"
+        step={0.1}
+        value={value2}
+      />
+    </Flex>
+  );
+}
+`}
+      />
+
+      <Example
+        id="ref example"
+        name="Example: ref"
+        description={`
+    Set a ref on NumberField to use the [Constraint validation API](https://developer.mozilla.org/en-US/docs/Web/Guide/HTML/Constraint_validation) or to anchor a Popover-based element.
```

Comment:
Can we explain a little bit more in here. From my observations, using the native mouse and buttons to increase/decrease the number will follow the max/min/steps. 

However, if entered manually outside the min/max range, it's still a valud entry and the only way to check for validity is ref.current.checkValidity()